int32 id
int32 indexTopNode 
int32 indexBottomNode
int32 topLayer
float32 value